app/jacoco_offline.gradle

```diff
@@ -0,0 +1,147 @@
+// https://stackoverflow.com/questions/53071373/powermock-jacoco-gradle-0-coverage-for-android-project?rq=1
+
+apply plugin: 'jacoco'
+
+configurations {
+    jacocoAnt
+    jacocoRuntime
+}
+
+jacoco {
+    toolVersion = "0.8.1"
+}
+
+def offline_instrumented_outputDir = "$buildDir.path/intermediates/classes-instrumented/debug"
+
+tasks.withType(Test) {
+    jacoco.includeNoLocationClasses = true
+}
+
+def coverageSourceDirs = [
+        'src/main/java'
+]
+
+task jacocoTestReport(type: JacocoReport, dependsOn: "test") {
+    group = "Reporting"
+    description = "Generate Jacoco coverage reports"
+
+    classDirectories = fileTree(
+            dir: 'build/intermediates/javac/debug/classes/',
+            excludes: ['**/R.class',
+                       '**/R$*.class',
+                       '**/BuildConfig.*',
+                       '**/MainActivity.*']
```

Comment:
This would exclude MainActivity from our coverage report.  I don't have such an activity.  Whoever wrote this guide must have needed to remove that file from their report.